doc/md/fundamentals/2-types/8-immutable-arrays.md

```diff
@@ -0,0 +1,193 @@
+- Performance optimization is needed: They provide efficient index access without dynamic resizing overhead.
+- Data integrity must be preserved: Immutability ensures that no accidental modifications occur.
+
+Immutable arrays do not allow in-place modifications, making them suitable when stability and performance are priorities.
+
+If the number of elements may change, collections like `List` or `Buffer` are a better choice, as immutable arrays require creating a new array each time an element is added or removed, which is inefficient.
+
+## Defining an immutable array
+
+Immutable arrays are declared using square brackets `[T]`. The type of the array is specified within the square brackets, e.g., `[Nat]` declares an immutable array of natural numbers.
+
+```motoko no-repl
+let immutableArray : [Nat] = [1, 2, 3, 4, 5];
+```
+
+## Accessing and modifying elements
+
+Attempting to access an array's index that does not exist will cause a [trap](https://internetcomputer.org/docs/motoko/fundamentals/basic-syntax/traps). Attempting to modify an immutable array will result in an error `expected mutable assignment target(M0073)`.
+
+```motoko no-repl
+let numbers : [Nat] = [10, 20, 30];
+
+let first : Nat = numbers[0];  // 10
+let second : Nat = numbers[1]; // 20
+
+Debug.print(debug_show(first));
+Debug.print(debug_show(second));
+```
+
+## Iterating through an array
+
+There are two primary ways to iterate through the elements in an array in Motoko:
+
+1. Using `array.values()`, which provides an iterator.
+
+2. Using a `for` loop that runs from `0` to `array.size() - 1`, as arrays are zero-based.
+
+Both methods achieve the same result, but `array.values()` is often preferred for its readability and simplicity.
+
+### Using `array.values()`
+
+The `array.values()` function returns an iterator that is used to iterate over the array's elements without manually managing indices.
+
+### Using a `for` loop
+
+A `for` loop can also be used to iterate over an array by accessing elements via their index.
+
+```motoko no-repl
+for (i in Iter.range(0, arr.size() - 1)) {
+    Debug.print(debug_show(arr[i]));
+}
+```
+
+## Converting an immutable array to a mutable array
+
+You can convert an immutable array into a mutable array using `Array.thaw` which is useful when working with data that needs to be modified in place after initially being immutable.
+
+```motoko no-repl
+let immutableArray : [Nat] = [1, 2, 3, 4, 5];
+
+let mutableCopy : [var Nat] = Array.thaw<Nat>(immutableArray);
+mutableCopy[0] := 10;
+//
+```
+
+
+## Passing a variable number of arguments
+
+Motoko supports passing collections to a function, ensuring that all arguments are handled as a collection rather than individual parameters.
+
+```motoko no-repl
+let greetings : [Text] = ["Hello, "Hola", "Ciao" ]
+
+func printAllStrings(strings : [Text]) {
+    for (s in strings.vals()) {
+        Debug.print(s);
+    }
+}
+```
```

Comment:
````suggestion

```motoko no-repl
func printAllStrings(strings : [Text]) {
    for (s in strings.values()) {
        Debug.print(s);
    }
};
printAllStrings(["Hello, "Hola", "Ciao" ]);
```
````